code/game/machinery/portable_turret.dm

```diff
@@ -503,35 +514,43 @@ var/list/turret_icons
 	set background = BACKGROUND_ENABLED
 
 	if(stat & (NOPOWER|BROKEN))
-		//if the turret has no power or is broken, make the turret pop down if it hasn't already
-		popDown()
+		if(!always_up)
+			//if the turret has no power or is broken, make the turret pop down if it hasn't already
+			popDown()
 		return
 
 	if(!enabled)
-		//if the turret is off, make it pop down
-		popDown()
+		if(!always_up)
+			//if the turret is off, make it pop down
+			popDown()
 		return
 
 	var/list/targets = list()			//list of primary targets
 	var/list/secondarytargets = list()	//targets that are least important
 
-	for(var/obj/mecha/ME in view(7,src))
+	for(var/obj/mecha/ME in view(scan_range,src))
 		assess_and_assign(ME.occupant, targets, secondarytargets)
 
-	for(var/obj/spacepod/SP in view(7,src))
+	for(var/obj/spacepod/SP in view(scan_range,src))
 		assess_and_assign(SP.pilot, targets, secondarytargets)
 
-	for(var/obj/vehicle/T in view(7,src))
+	for(var/obj/vehicle/T in view(scan_range,src))
 		assess_and_assign(T.buckled_mob, targets, secondarytargets)
 
-	for(var/mob/living/C in view(7,src))	//loops through all living lifeforms in view
+	for(var/mob/living/C in view(scan_range,src))	//loops through all living lifeforms in view

```

Comment:
no, byond caches view() calls if nothings changed and it caches for loop lists, so this is fine